### ISSUE TITLE
too much white space on top of captioned literal blocks in PDF output

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2460]
There are currently three cumulated vertical whitespace in PDF output for literal blocks with caption:
1. a `\literalblockcaptiontopvspace` which is half of `baselineskip`, typically 6pt,
2. a `\smallskip`, typically 3pt
3. a `\abovecaptionskip` coming from the caption itself, typically 10pt.

That makes for a total of about 20pt, this is a lot. Especially when the literal blocks is in an indented context.

Perhaps `\literalblocktopvspace` comes from a time when the caption was handled otherwise ? (perhaps put at the end of the code-blocks ?). I propose to suppress both `\literalblockcaptiontopvspace` and `\smallskip` (from the Verbatim code; it would be kept only for non-captioned case).

If this done the 

```
\newcommand{\literalblockneedspace}{5\baselineskip}
```

should use a smaller setting too, say `4\baselineskip`.

**Comment by jfbu** [comment]
For comparison here is output from current stable branch compared to output from current https://github.com/jfbu/sphinx/tree/latex_less_space_above_codecaptions (not yet a PR, as I am not sure myself if really better).

Current stable:

![testtopspace 5](https://cloud.githubusercontent.com/assets/2589111/14792219/956de128-0b19-11e6-8f0e-79fe1ea45ee9.png)

Now, with current https://github.com/jfbu/sphinx/tree/latex_less_space_above_codecaptions

![testtopspace-fix 5](https://cloud.githubusercontent.com/assets/2589111/14792232/a24b56dc-0b19-11e6-8f47-df450ff3bf0b.png)

Source:

``` rst
.. testtopspace documentation master file, created by
   sphinx-quickstart on Wed Apr 20 20:10:22 2016.
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

Welcome to testtopspace's documentation!
========================================

.. toctree::
   :maxdepth: 2


Section
-------


A first listing with a caption.

.. code-block:: python3
   :caption: Python 3.5 sample A
   :linenos:

    >>> 2 + 2
    4
    >>> 50 - 5*6
    20

A second listing but without a caption.

.. code-block:: python
   :linenos:

    >>> (50 - 5*6) / 4
    5.0
    >>> 8 / 5  # division always returns a floating point number
    1.6

..

   This third listing is quoted and with a caption

   .. code-block:: python
      :caption: Python 3.5 sample C
      :linenos:

       >>> 2 + 2
       4
       >>> 50 - 5*6
       20

   The fourth and fifth listings are in a list.

   * .. code-block:: python
         :caption: Python 3.5 sample D
         :linenos:

          >>> (50 - 5*6) / 4
          5.0
          >>> 8 / 5  # division always returns a floating point number
          1.6

   * .. code-block:: python
         :linenos:
         :caption: Python 3.5 sample E

          >>> (50 - 5*6) / 4
          5.0
          >>> 8 / 5  # division always returns a floating point number
          1.6

```

**Comment by tk0miya** [comment]
LGTM! Go ahead :-)

**Comment by jfbu** [comment]
fixed by b2b8fa41a59d4ac6a4327a6c126abcf3e0b7a44a
